### DISCUSSION TITLE
.bazelversion file

### DISCUSSION THREAD

**Discussion by manobaneru** [CodeIntelligenceTesting/jazzer#814] [General]
Please, can you help me? I don't know what I'm doing wrong but I can't find right version of bazel (from .bazelversion file)
![image](https://github.com/CodeIntelligenceTesting/jazzer/assets/72031386/a5b12233-b6b5-44eb-951a-c7d23942c7c7)
And if I try to use some older versions from the commits or edit  .bazelversion with something old I have:
- the same problem
- another error
![image](https://github.com/CodeIntelligenceTesting/jazzer/assets/72031386/3cfcbabd-5954-4f7d-9aec-6eadf1ce3e50)

**Comment by fmeum** [comment]
If you really want to build Jazzer from source (for example because you want to contribute to it), you can follow the steps in [CONTRIBUTING.md](https://github.com/CodeIntelligenceTesting/jazzer/blob/main/CONTRIBUTING.md). In particular, we recommend to install [Bazelisk](https://github.com/bazelbuild/bazelisk/releases/tag/v1.17.0) as `bazel` in your `PATH`, which will automatically download and use the correct version of Bazel for a given project.

That said, based on your screenshots, it looks like you are trying to use `rules_fuzzing`. For that, please see [its README section on Jazzer](https://github.com/bazelbuild/rules_fuzzing#java-fuzzing). You will have to add the `jazzer` config to your own project's `.bazelrc`, but any somewhat recent Bazel version should work. 

If you just want to try this out, you can also run the [Java examples](https://github.com/bazelbuild/rules_fuzzing/blob/master/examples/java/BUILD) directly from the `rules_fuzzing` repository, which comes with a predefined `jazzer` config:

```
$ git clone https://github.com/bazelbuild/rules_fuzzing
$ cd rules_fuzzing
$ bazel run --config=jazzer //examples/java:FuzzTest_run
```

(If you don't have `clang-10` installed, you may need to update the references to it in rules_fuzzing's `.bazelrc` with your local version first.)

**Reply by manobaneru** [reply]
Thank you so much for your answer!
